Tests/OpenCombineTests/PublisherTests/ReplaceEmptyTests.swift

```diff
@@ -0,0 +1,277 @@
+                                                     .requested(.unlimited)])
+        XCTAssertEqual(helper.tracking.history, [.subscription("ReplaceEmpty"),
+                                                 .value(33),
+                                                 .completion(.finished)])
+
+        helper.publisher.send(completion: .failure(.oops))
+        helper.publisher.send(completion: .failure(.oops))
+        XCTAssertEqual(helper.tracking.history, [.subscription("ReplaceEmpty"),
+                                                 .value(33),
+                                                 .completion(.finished)])
+    }
+
+    func testCancelAlreadyCancelled() throws {
+        let helper = OperatorTestHelper(publisherType: CustomPublisher.self,
+                                        initialDemand: .unlimited,
+                                        receiveValueDemand: .none,
+                                        createSut: { $0.replaceEmpty(with: -7) })
+
+        try XCTUnwrap(helper.downstreamSubscription).cancel()
+        try XCTUnwrap(helper.downstreamSubscription).request(.unlimited)
+        try XCTUnwrap(helper.downstreamSubscription).cancel()
+
+        XCTAssertEqual(helper.subscription.history, [.requested(.unlimited),
+                                                     .requested(.unlimited),
+                                                     .cancelled])
+
+        helper.publisher.send(completion: .failure(.oops))
+        helper.publisher.send(completion: .finished)
+
+        XCTAssertEqual(helper.subscription.history, [.requested(.unlimited),
+                                                     .requested(.unlimited),
+                                                     .cancelled])
+        XCTAssertEqual(helper.tracking.history, [.subscription("ReplaceEmpty")])
+    }
+
+    func testReceiveSubscriptionTwice() throws {
+        let helper = OperatorTestHelper(
+            publisherType: CustomPublisher.self,
+            initialDemand: nil,
+            receiveValueDemand: .none,
+            createSut: { $0.replaceEmpty(with: 22) }
+        )
+
+        XCTAssertEqual(helper.subscription.history, [.requested(.unlimited)])
+
+        let secondSubscription = CustomSubscription()
+
+        try XCTUnwrap(helper.publisher.subscriber)
+            .receive(subscription: secondSubscription)
+
+        XCTAssertEqual(secondSubscription.history, [.cancelled])
+
+        try XCTUnwrap(helper.publisher.subscriber)
+            .receive(subscription: helper.subscription)
+
+        XCTAssertEqual(helper.subscription.history, [.requested(.unlimited), .cancelled])
+
+        try XCTUnwrap(helper.downstreamSubscription).cancel()
+
+        XCTAssertEqual(helper.subscription.history, [.requested(.unlimited),
+                                                     .cancelled,
+                                                     .cancelled])
+    }
```

Comment:
Could you leverage the existing function for testing this declared in [CommontTests.swift](https://github.com/broadwaylamb/OpenCombine/blob/899a04bb3fc88a855196a0ce90d52b6f1cf5e04c/Tests/OpenCombineTests/Helpers/CommonTests.swift#L140-L167)?

Also, I'd like to ask you to add the following tests from that file:

- `testReceiveValueBeforeSubscription`
- `testReceiveCompletionBeforeSubscription`
- `testRequestBeforeSubscription`
- `testCancelBeforeSubscription`

For example usage you can refer to [this](https://github.com/broadwaylamb/OpenCombine/blob/899a04bb3fc88a855196a0ce90d52b6f1cf5e04c/Tests/OpenCombineTests/PublisherTests/DelayTests.swift#L417-L447).

Also, there's one more test missing: `testLifecycle` from [TestLifecycle.swift](https://github.com/broadwaylamb/OpenCombine/blob/899a04bb3fc88a855196a0ce90d52b6f1cf5e04c/Tests/OpenCombineTests/Helpers/TestLifecycle.swift#L17-L153). Example [here](https://github.com/broadwaylamb/OpenCombine/blob/899a04bb3fc88a855196a0ce90d52b6f1cf5e04c/Tests/OpenCombineTests/PublisherTests/HandleEventsTests.swift#L232-L236).

Reply:
@broadwaylamb In this case, there was a different behavior for the receives subscription twice than the usual: 

```
CommonTests.swift:150: error: -[OpenCombineTests.ReplaceEmptyTests testReceiveSubscriptionTwice] : XCTAssertEqual failed: ("[.requested(.unlimited)]") is not equal to ("[]")
CommonTests.swift:162: error: -[OpenCombineTests.ReplaceEmptyTests testReceiveSubscriptionTwice] : XCTAssertEqual failed: ("[.requested(.unlimited), .cancelled]") is not equal to ("[.cancelled]")
CommonTests.swift:166: error: -[OpenCombineTests.ReplaceEmptyTests testReceiveSubscriptionTwice] : XCTAssertEqual failed: ("[.requested(.unlimited), .cancelled, .cancelled]") is not equal to ("[.cancelled, .cancelled]")
```

I will add the other following tests from the file though.

Reply:
Are you experiencing this behavior with Combine too?

Reply:
Oh, I see. Okay then, let's leave this test as-is.

Reply:
I am experiencing it when running the tests with the compatibility flag.